### ISSUE TITLE
LSTM Peephole Connections backpropagation

### ISSUE THREAD

**Issue by Pozimek** [pytorch/pytorch#28973]
Feel free to correct me if my understanding of this issue is wrong, as I feel pretty confused about it myself. [The LSTM paper](ftp://ftp.idsia.ch/pub/juergen/TimeCount-IJCNN2000.pdf) defines a [specific rule](https://i.imgur.com/peOKqkL.png) for gradient updates of the 'peephole' connections that enable each of the gates to inspect the memory cell. The writing style is quite painful, but the part I'm referring to is:

> [...] during learning no error signals are propagated back from gates via peephole connections to CEC

However, the peephole connections are also weighted and thus should be subject to gradient updates. It's just that the gradient that updates them should not flow any further into the CEC. How would a forward pass that implements this feature look like? Using .detach() on the peephole weights is not desirable as they will not be updated that way. Would it make sense to call .detach() on the memory cell itself when multiplying with the peephole weights?

cc @ezyang @zou3519

**Comment by ezyang** [comment]
As a question on usage of PyTorch, you will probably get better answers on https://discuss.pytorch.org/ However, you may be able to implement the behavior you're looking for using backward hooks that, for example, zero out gradients in directions you don't want them to flow.
